src/chocolatey.resources/helpers/functions/Get-Installed.ps1

```diff
@@ -0,0 +1,46 @@
+﻿## Get-Installed
+##############################################################################################################
+## Returns a list of installed applications.
+## Registry based replacement for "Get-WmiObject -Class Win32_Product".
+##############################################################################################################
+
+function Get-Installed {
+param(
+  [string] $NonMSIOption = 'HideNonMSI' # You may use 'ShowNonMSI' option
+)
+
+  if(($NonMSIOption -ine 'HideNonMSI') -and ($NonMSIOption -ine 'ShowNonMSI')) {
+    throw "Unsupported argument: $NonMSIOption"
+  }
+
+
+  $items = @()
+
+  if(Test-Path "HKLM:SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall") {
+    $items += Get-ChildItem "HKLM:SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall"
+  }
+  if(Test-Path "HKLM:SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Uninstall") {
+    $items += Get-ChildItem "HKLM:SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Uninstall"
+  }
+
+  $Win32_Product = @()
+
+  foreach ($app in $items)
+  {
+    $propitems = $app | Foreach-Object {Get-ItemProperty $_.PsPath}
+    $w32item = New-Object System.Object
+    $w32item | Add-Member -type NoteProperty -Name "IdentifyingNumber" -Value $app.PsChildname
+    $w32item | Add-Member -type NoteProperty -Name "Name"    -Value $propitems.DisplayName
+    $w32item | Add-Member -type NoteProperty -Name "Version" -Value $propitems.DisplayVersion
+    $w32item | Add-Member -type NoteProperty -Name "Vendor"  -Value $propitems.Publisher
+    $w32item | Add-Member -type NoteProperty -Name "Caption" -Value $propitems.DisplayName
+
+    if($w32item.Name) {
+      if(($NonMSIOption -ieq 'ShowNonMSI') -or ($propitems.WindowsInstaller -eq "1")) {
+        $Win32_Product += $w32item
+      }
+    }
+  }
+
+  return $Win32_Product

```

Comment:
The only difference is properties order. This doesn't play a role as far as I understand.